fabric-message-api-v1/src/client/java/net/fabricmc/fabric/api/message/v1/ClientMessageEvents.java

```diff
@@ -0,0 +1,301 @@
+	 * To toggle overlay, return false and call
+	 * {@link net.minecraft.client.network.ClientPlayerEntity#sendMessage(Text, boolean) ClientPlayerEntity.sendMessage(message, overlay)}.
+	 */
+	public static final Event<AllowReceiveGameMessage> ALLOW_RECEIVE_GAME_MESSAGE = EventFactory.createArrayBacked(AllowReceiveGameMessage.class, listeners -> (message, overlay) -> {
+		for (AllowReceiveGameMessage listener : listeners) {
+			if (!listener.allowReceiveGameMessage(message, overlay)) {
+				return false;
+			}
+		}
+
+		return true;
+	});
+
+	/**
+	 * An event triggered when the client receives a chat message,
+	 * which is any message sent by a player. Is not called when
+	 * {@linkplain #ALLOW_RECEIVE_CHAT_MESSAGE chat messages are blocked}.
+	 */
+	public static final Event<ReceiveChatMessage> RECEIVE_CHAT_MESSAGE = EventFactory.createArrayBacked(ReceiveChatMessage.class, listeners -> (message, signedMessage, sender, params, receptionTimestamp) -> {
+		for (ReceiveChatMessage listener : listeners) {
+			listener.onReceiveChatMessage(message, signedMessage, sender, params, receptionTimestamp);
+		}
+	});
+
+	/**
+	 * An event triggered when the client receives a chat message,
+	 * which is any message sent by the server. Is not called when
+	 * {@linkplain #ALLOW_RECEIVE_CHAT_MESSAGE chat messages are blocked}.
+	 *
+	 * <p>Overlay is whether the message will be displayed in the action bar.
+	 * Use {@link #ALLOW_RECEIVE_GAME_MESSAGE to toggle overlay}.
+	 */
+	public static final Event<ReceiveGameMessage> RECEIVE_GAME_MESSAGE = EventFactory.createArrayBacked(ReceiveGameMessage.class, listeners -> (message, overlay) -> {
+		for (ReceiveGameMessage listener : listeners) {
+			listener.onReceiveGameMessage(message, overlay);
+		}
+	});
+
+	@FunctionalInterface
+	public interface AllowSendChatMessage {
+		/**
+		 * Called when the client is about to send a chat message,
+		 * typically from a client GUI. Returning {@code false}
+		 * prevents the message from being sent and the {@link #SEND_CHAT_MESSAGE} event
+		 * from triggering.
+		 *
+		 * @param message the message that will be sent to the server
+		 * @return {@code true} if the message should be sent, otherwise {@code false}
+		 */
+		boolean allowSendChatMessage(String message);
+	}
+
+	@FunctionalInterface
+	public interface AllowSendCommandMessage {
+		/**
+		 * Called when the client is about to send a command,
+		 * which is whenever the player executes a command
+		 * including client commands registered with {@code fabric-command-api}.
+		 * Returning {@code false} prevents the command from being sent and
+		 * the {@link #SEND_COMMAND_MESSAGE} event from triggering.
+		 *
+		 * <p>Also, this is called before client commands are checked and executed.
+		 * Therefore, client commands registered with {@code fabric-command-api}
+		 * will not be executed if {@code false} is returned.
+		 *
+		 * @param message the command that will be sent to the server
```

Comment:
Does this have the initial slash? I feel that's a good thing to document somewhere